src/stream/src/executor/dispatch.rs

```diff
@@ -110,56 +112,63 @@ struct DispatchExecutorInner {
 }
 
 impl DispatchExecutorInner {
-    async fn dispatch(&mut self, msg: Message) -> StreamResult<()> {
+    async fn dispatch(&mut self, msg: MessageBatch) -> StreamResult<()> {
         let limit = (self.context.config.developer).exchange_concurrent_dispatchers;
-
+        // Only barrier can be batched for now.
```

Comment:
Not strongly related to this PR: I wonder whether peaking the message including barrier/chunk and batch them by size into one gRPC message can be a general optimization to amortize network cost.

Theoretically it is http2 and tcp under the hook so the network package should be batched already but given that the barrier batch optimization works, I wonder whether there are other overheads that we are not aware of in gRPC.

I also saw a few asks for gRPC streaming rpc message batching elsewhere:
- https://github.com/grpc/grpc-go/issues/524
- https://github.com/grpc/grpc-dotnet/issues/2074

Reply:
> but given that the barrier batch optimization works

I think this PR primarily reduces the total number of poll from `RemoteInput` and `LocalInput` in `MergeExecutor`.

Reply:
I believe the bottleneck lies in the [SelectReceivers](https://github.com/risingwavelabs/risingwave/blob/366a139fe42de6bf2e7d239d7ac7ea14c054a97c/src/stream/src/executor/merge.rs#L388), which aligns multiple inputs.